students/DennisLee/lesson02/grid_printer_class.py

```diff
@@ -0,0 +1,78 @@
+"Print a table grid."
+
+
+class TableGrid(object):
+    """
+    Create a square table grid, specifying the number of rows/columns
+    and the cell height/width.
+
+    :rows_and_cols:  The number of rows and columns in the grid.
+
+    :cell_size:  The height and width of a grid cell.
+    """
+
+    def __init(self, rows_and_cols: int, cell_size: int):
+        self.rows_and_cols = rows_and_cols
+        self.cell_size = cell_size
+
+    def print_grid(self) -> bool:
+        """
+        Print a square table grid using plain text characters.
+
+        :return:  True if the function succeeds; False if the function fails.
+        """
+
+        horizontal_rule: str = self.build_line(True)
+        interior_line: str = self.build_line(False)
+
+        print((horizontal_rule + (interior_line * cell_size)) * rows_and_cols
+             + horizontal_rule)
+        return True
+
+    def build_line(self, border: bool = False) -> str:
+        """
+        Build a single horizontal line within the table grid.
+
+        :border:  Whether to create a horizontal rule. True indicates yes;
+                  False (default) specifies an interior line (containing just
+                  the vertical rules, with spaces in between those borders for
+                  the cell interior).
+
+        :return:  The text string for a table line.
+        """
+
+        PLUS, MINUS, PIPE, SPACE, LINEFEED = '+', '-', '|', ' ', '\n'
```

Comment:
typically constants should go at the top of the file, otherwise they should be regular lower case vars when within a method.